tunnel/forwarder_test.go

```diff
@@ -0,0 +1,6 @@
+package tunnel
+
+import "testing"
+
+func TestBidirectionalPipeline_Run(t *testing.T) {
```

Comment:
Did you mean to add tests?

Reply:
Yeah. This is tested implicitly through the `tunnel_normal_test.go` and `tunnel_reverse_test.go` files, but we could use some unit tests here too.